staging/src/k8s.io/api/node/v1alpha1/types.go

```diff
@@ -114,3 +114,76 @@ type RuntimeClassList struct {
 	// Items is a list of schema objects.
 	Items []RuntimeClass `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +k8s:conversion-gen:explicit-from=net/url.Values
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeLogViewerOptions is the query options for a Node's logs REST call. The options specified here intersect unless
+// called out otherwise.
+type NodeLogViewerOptions struct {
```

Comment:
this type is being added to a v1alpha1 package, but the query param decoding is not versioned... I think it is misleading to add a struct here that we don't actually have the ability to version... I think this should be added to a v1 package (either to core/v1 alongside `Node`, like `PodLogOptions`, or to node/v1 alongside other node-related types) to make it clear that we don't have the ability to evolve the type in incompatible ways

Reply:
This is a great comment, and shows me how little I know about how the pass-down to Kubelet actually works.  Even putting it in v1 is suspect, since (in theory) we could have a v2 and we can't really change this, unless we version it at kubelet. 

Instead of kubelet answering on `/logs/`, should it be `/v1alpha1/logs` ?  I guess the apiserver and kubelet would need some version negotiation or that becomes a laggard within the skew window.  I know this is sort of net-new.  The alternative is that these params are crystallized ~forever, so we'd better be REALLY sure about them. IOW, less is more.

Reply:
@liggitt @thockin I am restarting work on this. So should I be moving `NodeLogViewerOptions` to a v1 package?

Reply:
I don't have a lot of hands-on with how we have managed other such Options types.  It seems like not having a way to do this will eventually bite us in the butt, but short term I guess I would go with Jordan's suggestion of core/v1 or node/v1 (whichever you think makes more sense - my guess is core)

Reply:
Thanks @thockin. I have gone with `core/v1`.